src/invert/laplacexy/laplacexy.cxx

```diff
@@ -113,20 +164,187 @@ LaplaceXY::LaplaceXY(Mesh *m, Options *opt, const CELL_LOC loc)
   PetscMalloc( (localN)*sizeof(PetscInt), &d_nnz );
   PetscMalloc( (localN)*sizeof(PetscInt), &o_nnz );
   
+  if (finite_volume) {
+    setPreallocationFiniteVolume(d_nnz, o_nnz);
+  } else {
+    setPreallocationFiniteDifference(d_nnz, o_nnz);
+  }
+  // Pre-allocate
+  MatMPIAIJSetPreallocation( MatA, 0, d_nnz, 0, o_nnz );
+  MatSetUp(MatA); 
+  
+  PetscFree( d_nnz );
+  PetscFree( o_nnz );
+  
+  // Determine which row/columns of the matrix are locally owned
+  int Istart, Iend;
+  MatGetOwnershipRange( MatA, &Istart, &Iend );
+  
+  // Convert indexXY from local index to global index
+  indexXY += Istart;
+  
+  // Now communicate to fill guard cells
+  localmesh->communicate(indexXY);
+
+  // Fill corner guard cells by getting the y-guard cells from the x-neighbours.
+  // This is an arbitrary choice, which makes a difference at the X-point, where taking
+  // the x-guard cells from the y-neighbours would give points in different regions.
+  auto xcomm = localmesh->getXcomm();
```

Comment:
Can remove all this communication of corners now that #1885 has gone in. I'll push an update...